src/math/iir_df2t_generic.c

```diff
@@ -104,4 +104,24 @@ int32_t iir_df2t(struct iir_state_df2t *iir, int32_t x)
 	return out;
 }
 
+int16_t iir_df2t_s16(struct iir_state_df2t *iir, int16_t x)
+{
+	return sat_int16(Q_SHIFT_RND(iir_df2t(iir, ((int32_t)x) << 16), 31, 15));
+}
```

Comment:
These should all be static inline in the header.

Reply:
Sure, need to think how to make in a nice way the generic vs. hifi3 headers. Here I was hoping compiler does the inline but it's of course not guaranteed.

Reply:
not when this is an archive. you can do
```c
/* the header file */
#if USE_THE_STATIC_INLINE_VERSION
static inline int16_t iir_df2t_s16(struct iir_state_df2t *iir, int16_t x)
{
	return sat_int16(Q_SHIFT_RND(iir_df2t(iir, ((int32_t)x) << 16), 31, 15));
}
#else
/* just declare func */
int16_t iir_df2t_s16(struct iir_state_df2t *iir, int16_t x);
#endif  
```

Reply:
@lgirdwood I'm testing approach to have in iir_df2t.h next code chunk.

```
/* Inline functions with or without HiFi3 intrinsics */
#if IIR_HIFI3
#include "iir_df2t_hifi3.h"
#else
#include "iir_df2t_generic.h"
#endif

```

The inline functions e.g. in iir_df2t_hifi3.h are like:

```
static inline int16_t iir_df2t_s16(struct iir_state_df2t *iir, int16_t x)
{
	ae_f32x2 y = iir_df2t(iir, ((int32_t)x) << 16);

	return AE_ROUND16X4F32SSYM(y, y);
}

```

Is such OK?

Reply:
should be , but what is the ```iir_df2t``` definition ?

Reply:
It's https://github.com/thesofproject/sof/blob/main/src/math/iir_df2t_generic.c or https://github.com/thesofproject/sof/blob/main/src/math/iir_df2t_hifi3.c depending on build.